base/src/main/java/org/bitcoinj/base/Difficulty.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.base;
+
+import org.bitcoinj.base.internal.ByteUtils;
+
+import java.math.BigInteger;
+import java.time.Duration;
+import java.util.Objects;
+
+import static org.bitcoinj.base.internal.Preconditions.checkArgument;
+
+/**
+ * The target difficulty is a value which a header hash must be equal to or below in order for that header to be a
+ * valid part of the block chain.
+ * <p>
+ * Difficulty values cannot be negative. A high value means low difficulty, and a low value means high difficulty.
+ * <p>
+ * The value is stored in only 32 bits of space, so it uses a less precise “compact form”, or "nBits". Think of it as a
+ * base-256 version of the scientific notation, consisting of a 1 byte exponent and 3 bytes of mantissa. That form is
+ * used for storing the value in this class as well, so when constructing from an integer value, be prepared to lose
+ * precision.
+ * <p>
```

Comment:
Could we provide a few examples of compact format here?  (Alternatively we could link to some of the static constants as examples)

Reply:
The best explanation I've seen is https://learnmeabitcoin.com/technical/block/bits/, but do we want to link to such a site?

Reply:
But I added a link to one of the constants: 90aa14c67

Reply:
> do we want to link to such a site?

Probably not since it is "unofficial" and could disappear or get squatted.

I think the added link is good enough.